spec/acceptance/docker_full_spec.rb

```diff
@@ -0,0 +1,643 @@
+require 'spec_helper_acceptance'
+
+describe 'Temp OS hacks' do
+  if fact('operatingsystem') =~ /CentOS/
+    if fact('operatingsystemmajrelease') == '7'
+        pp=<<-EOS
+              package {'device-mapper':
+                ensure => latest,
+              }
+        EOS
+      apply_manifest(pp, :catch_failures => true)
+    end
+  elsif default['platform'] =~ /el-7/
+      pp=<<-EOS
+            package {'device-mapper':
+              ensure => latest,
+            }
+
+            package {'docker':
+              install_options => ['--enablerepo=rhel7-extras'],
+              ensure => latest,
+              require => Package['device-mapper']
+            }
+      EOS
+      apply_manifest(pp, :catch_failures => true)
+  end
+end
+
+describe 'docker full acceptance tests section 1 - testing class {\'docker\':}' do
+  context 'test 1.1 - Install and configure Docker and Docker Daemon' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': ps -aux should be running docker process' do
+      shell('ps -aux | grep docker') do |r|
+        expect(r.stdout).to match(/\/usr\/bin\/docker/)
+      end
+    end
+    it ': docker ps should output table headers' do
+      shell('docker ps') do |r|
+        expect(r.stdout).to match(/CONTAINER ID\s+IMAGE\s+COMMAND\s+CREATED\s+STATUS\s+PORTS\s+NAMES/)
+      end
+    end
+  end 
+
+  context 'test 1.3a - Bound to a particular TCP socket' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': 
+          tcp_bind    => 'tcp://127.0.0.1:4444',
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': netstat should show docker listening' do
+      shell('netstat -tulpn | grep docker') do |r|
+        expect(r.stdout).to match(/tcp\s+0\s+0\s+127.0.0.1:4444\s+0.0.0.0\:\*\s+LISTEN\s+\d+\/docker/)
+      end
+    end
+    it ': docker ps should output the table headers' do
+      shell('docker ps') do |r|
+        expect(r.stdout).to match(/CONTAINER ID\s+IMAGE\s+COMMAND\s+CREATED\s+STATUS\s+PORTS\s+NAMES/)
+      end
+    end
+  end  
+
+  context 'test 1.3b - Bound to a particular unix socket' do
```

Comment:
I don't like the test 1.3b part, it feels like it could refer to something that's not part of the repo. If you could remove from all the tests.